opencart1.5.x/upload/catalog/controller/payment/xendit.php

```diff
@@ -49,9 +49,14 @@ public function process_payment() {
 
         try {
             $response = Xendit::request($request_url, Xendit::METHOD_POST, $request_payload, $request_options);
-            
+
             if (isset($response['error_code'])) {
-                $json['error'] = $response['message'];
+                $message = $response['message'];
+
+                if (isset($response['code'])) {
+                    $message .= ". Code: " . $response['code'];
```

Comment:
i think all the message already end with a period. So you can omit that when appending the Code: XXXXXX